lib/modules/manager/npm/post-update/npm.ts

```diff
@@ -251,7 +251,7 @@ export function divideWorkspaceAndRootDeps(
         // compare workspaceDir to workspace patterns
         // stop when the first match is found and
         // add workspaceDir to workspaces set and upgrade object
-        for (const workspacePattern of workspacePatterns ?? []) {
+        for (const workspacePattern of workspacePatterns) {
```

Comment:
```suggestion
        for (const workspacePattern of coerceArray(workspacePatterns)) {
```
better be defense, this can be undefined

Reply:
This statement is inside an if-block which already validates `workspacePatterns` so this won't be necessary.
```ts
Line 237-241: 
if (
      upgrade.managerData.workspacesPackages?.length &&
      is.string(upgrade.packageFile)
    ) {
      const workspacePatterns = upgrade.managerData.workspacesPackages;
```